### ISSUE TITLE
Support configurable chunk path prefix

### ISSUE THREAD

**Issue by ggoodman** [evanw/esbuild#733]
Given a configuration like this:

```js
{
  bundle: true,
  entrypoints: ['./src/A/index.ts', './src/B/index.ts'],
  format: 'esm',
  outdir: './dist',
  outbase: './src',
  splitting: true,
}
```

The resulting build might look like this:

```
dist
  A
    index.js
  B
    index.js
  chunk.1.js
  chunk.2.js
```

If a `package.json` is put in that directory and it is used as the basis of producing an npm module called `my-module`, tooling like the TypeScript language server will complete `import {} from 'my-module/` with a dialogue showing all of the chunks intermingle with the legitimate sub-packages (at least until it starts respecting the `"exports"` field in the `package.json`).

It would be helpful to have a measure of control over where the chunks get placed to prevent them from polluting completion in this way. It's a minor thing but perhaps a bit of feedback that might be helpful in your planned code-splitting refactor.

**Comment by evanw** [comment]
Closing as a duplicate of #553 and #187.

**Comment by evanw** [comment]
FYI this is being implemented ahead of #553. See #888 and the release notes for more information.
